tests/actions/notifications/index.test.ts

```diff
@@ -0,0 +1,216 @@
+
+      MockDate.set(timeout + 100)
+      jest.runAllTimers()
+      expect(mockStore.getActions()).toMatchSnapshot()
+    })
+
+    it('should change to the next notification immediately if active is non-dismissible', () => {
+      const sticky = newSticky()
+      const slidy = newSlidy()
+      slidy.dismiss = { timeout }
+
+      mockStore.dispatch(scheduleNotification(sticky))
+      mockStore.dispatch(scheduleNotification(slidy))
+      expect(mockStore.getActions()).toMatchSnapshot()
+
+      MockDate.set(timeout + 100)
+      jest.runAllTimers()
+      expect(mockStore.getActions()).toMatchSnapshot()
+    })
+
+    it('should not change to a new non-dismissible if active is dismissible', () => {
+      const sticky = newSticky()
+      const slidy = newSlidy()
+      slidy.dismiss = { timeout }
+
+      mockStore.dispatch(scheduleNotification(slidy))
+      mockStore.dispatch(scheduleNotification(sticky))
+      expect(mockStore.getActions()).toMatchSnapshot()
+
+      MockDate.set(timeout + 100)
+      jest.runAllTimers()
+      expect(mockStore.getActions()).toMatchSnapshot()
+    })
+  })
+
+  describe('invokeDismiss', () => {
+    beforeEach(() => {
+      mockStore.reset()
+      MockDate.set(0)
+    })
+    afterEach(() => MockDate.reset())
+
+    it('should call the onDismiss callback and remove the notification', () => {
+      const onDismiss = jest.fn()
+      const slidy = newSticky({
+        dismiss: { onDismiss },
+      })
+
+      mockStore.dispatch(scheduleNotification(slidy))
+      expect(mockStore.getActions()).toMatchSnapshot()
+      mockStore.dispatch(invokeDismiss(slidy))
+      jest.runAllTimers()
+      expect(mockStore.getActions()).toMatchSnapshot()
+      expect(onDismiss).toHaveBeenCalledTimes(1)
+    })
+  })
+
+  describe('invokeInteract', () => {
+    beforeEach(() => {
+      mockStore.reset()
+      MockDate.set(0)
+    })
+    afterEach(() => MockDate.reset())
+
+    it('should call the onInteract callback and remove the notification', () => {
+      const onInteract = jest.fn()
+      const sticky = newSticky({
+        interact: {
+          label: 'DO IT',
+          onInteract,
+        },
+      })
+
+      mockStore.dispatch(scheduleNotification(sticky))
+      expect(mockStore.getActions()).toMatchSnapshot()
+      mockStore.dispatch(invokeInteract(sticky))
+      jest.runAllTimers()
+      expect(mockStore.getActions()).toMatchSnapshot()
+      expect(onInteract).toHaveBeenCalledTimes(1)
+    })
+
+    it('should keep the notification if onInteract returns true', () => {
+      const onInteract = jest.fn().mockReturnValue(true)
+      const sticky = newSticky({
+        interact: {
+          label: 'DO IT',
+          onInteract,
+        },
+      })
+
+      mockStore.dispatch(scheduleNotification(sticky))
+      expect(mockStore.getActions()).toMatchSnapshot()
+      mockStore.dispatch(invokeInteract(sticky))
+      expect(mockStore.getActions()).toMatchSnapshot()
```

Comment:
Should this run after the timers have ran? I.e. swapping this and the line below.

Reply:
If not, then what does `runAllTimers()` do here?

Reply:
it should be run before and after, to ensure things haven't changed even after scheduled updates may have run

Reply:
I fixed it to run after as well